internal/kafka/internal/presenters/metric_test.go

```diff
@@ -0,0 +1,312 @@
+			args: args{
+				from: mocks.GetNotAllowedSampleStream(),
+			},
+			want: mocks.GetNotAllowedLabelRangeQuery(),
+		},
+		{
+			name: "should return RangeQuery converted from SampleStream with supported label",
+			args: args{
+				from: mocks.GetSampleStream(),
+			},
+			want: mocks.GetRangeQueryStringMapMetric(),
+		},
+	}
+
+	RegisterTestingT(t)
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			Expect(convertSampleStream(tt.args.from)).To(Equal(tt.want))
+		})
+	}
+}
+
+func TestConvertSample(t *testing.T) {
+	type args struct {
+		from *pModel.Sample
+	}
+
+	tests := []struct {
+		name string
+		args args
+		want public.InstantQuery
+	}{
+		{
+			name: "should return InstantQuery converted from Sample with not supported label",
+			args: args{
+				from: mocks.GetNotAllowedLabelsSample(),
+			},
+			want: mocks.GetSampleInstantQueryNoMetrics(),
+		},
+		{
+			name: "should return InstantQuery converted from Sample with supported label",
+			args: args{
+				from: mocks.GetSampleWithValidLabel(),
+			},
+			want: mocks.GetInstantQueryWithMetrics(),
+		},
+	}
+
+	RegisterTestingT(t)
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			Expect(convertSample(tt.args.from)).To(Equal(tt.want))
+		})
+	}
+}
+
+func TestConvertSamplePair(t *testing.T) {
+	type args struct {
+		from *pModel.SamplePair
+	}
+
+	tests := []struct {
+		name string
+		args args
+		want public.Values
+	}{
+		{
+			name: "should successfully convert to Values from non-empty SamplePair",
+			args: args{
+				from: mocks.GetSamplePair(),
+			},
+			want: mocks.GetSampleValue(),
+		},
+	}
+
+	RegisterTestingT(t)
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			Expect(convertSamplePair(tt.args.from)).To(Equal(tt.want))
+		})
+	}
+}
+
+func TestPresentMetricsByRangeQuery(t *testing.T) {
+	type args struct {
+		metrics *observatorium.KafkaMetrics
+	}
+
+	tests := []struct {
+		name    string
+		args    args
+		want    []public.RangeQuery
+		wantErr bool
+	}{
+		{
+			name: "should throw an error if attempting to present RangeQuery slice with metrics containing error",
+			args: args{
+				metrics: mocks.GetKafkaMetricWithError(),
+			},
+			wantErr: true,
+		},
+		{
+			name: "should present RangeQuery slice when using valid metrics object",
+			args: args{
+				metrics: mocks.GetSampleKafkaMetric(),
+			},
+			wantErr: false,
+			want:    mocks.GetSampleRangeQuerySlice(),
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			firstQuery, err := PresentMetricsByRangeQuery(tt.args.metrics)
+			if !tt.wantErr && err != nil {
+				t.Errorf("unexpected error for PresentMetricsByRangeQuery: %v", err)
+			}
+			if !tt.wantErr {
+				firstWantQuery := tt.want[0]
+				if firstQuery[0].Metric["__name__"] != firstWantQuery.Metric["__name__"] {
+					t.Errorf("unexpected result for PresentMetricsByRangeQuery - want: %+v got: %+v", tt.want, firstQuery)
+				}
+			}
+		})
```

Comment:
We can use `gomega.Expect...` for these too

Reply:
thanks Manyanda. I have fixed that too